test/cutting/test_cutting_roundtrip.py

```diff
@@ -142,8 +142,9 @@ def test_cutting_exact_reconstruction(example_circuit):
     subcircuits, bases, subobservables = partition_problem(
         qc, "AAB", observables=observables_nophase
     )
+    qpu = [None, "eagle", "heron"][np.random.randint(2)]
```

Comment:
Unit tests are fine to make sure the equivalences can be accessed and used in the ways we intend, but the equivalence bugs that worry me are bugs involving indexing and expval correctness, so I really want to fold the equivalence testing into the roundtrip tests.

That being said, I didn't want to double the runtime of this test by making a whole new loop testing the equivalences, so I will just randomly translate subexperiments and make sure we can reconstruct the exact expval.

I will sample the standard gate set 50% of the time and the eagle/heron gateset 50% of the time